### PR TITLE
Completed express-notes-api challenge. - Daniel Tak

### PR DESCRIPTION
**Pull request by d-tak**
**GET**
Screen Shot 2022-08-23 at 9 18 24 PM
Screen Shot 2022-08-23 at 9 19 20 PM
Screen Shot 2022-08-23 at 9 19 44 PM
Screen Shot 2022-08-23 at 9 20 05 PM

### CODE REVIEW

**Comment by d-tak** [comment]
**POST**
Screen Shot 2022-08-22 at 9 21 45 PM
Screen Shot 2022-08-23 at 9 20 54 PM
Screen Shot 2022-08-23 at 9 21 57 PM

**Comment by d-tak** [comment]
**DELETE**
Screen Shot 2022-08-23 at 9 23 50 PM
Screen Shot 2022-08-23 at 9 24 09 PM
Screen Shot 2022-08-23 at 9 24 44 PM

**Comment by d-tak** [comment]
**PUT**
Screen Shot 2022-08-23 at 9 16 46 PM
Screen Shot 2022-08-23 at 9 16 17 PM
Screen Shot 2022-08-23 at 9 15 56 PM
Screen Shot 2022-08-23 at 9 15 34 PM
Screen Shot 2022-08-23 at 9 15 19 PM

File: express-notes-api/index.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  } else {
+    data.notes[replaceById].content = req.body.content;
+    fs.writeFile('data.json', JSON.stringify(data, null, 2), err => {
+      if (err) {
+        console.error(err);
+        res.status(500);
+        res.json({ Error: 'An unexpected error occurred.' });
+      } else {
+        res.sendStatus(201);
+        res.json(data.notes[replaceById]);
```
Comment (TimelikeClosure) [inline express-notes-api/index.js:94]: Please make sure you're not sending two responses to the same request.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Comment by d-tak** [comment]
Screen Shot 2022-08-24 at 10 18 40 AM

File: express-notes-api/index.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  } else {
+    data.notes[replaceById].content = req.body.content;
+    fs.writeFile('data.json', JSON.stringify(data, null, 2), err => {
+      if (err) {
+        console.error(err);
+        res.status(500);
+        res.json({ Error: 'An unexpected error occurred.' });
+      } else {
+        res.sendStatus(200);
+      }
```
Comment (TimelikeClosure) [inline express-notes-api/index.js:94]: Successful `PUT /api/notes/:id` requests _should_ get a response body back with the updated note:

---

image

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**